### PR TITLE
chore: make generics optional

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: [AB#33405](https://statoil-proview.visualstudio.com/9949ad5e-7d15-4531-901e-296574079ee1/_workitems/edit/33405)

## Type of change

Remove options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Refactor (A code change that neither fixes a bug nor adds a feature)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read.
-   [ ] Documentation and comments is added where needed.
-   [ ] My code is covered by tests.
-   [ ] UX has reviewed relevant UI contributions.

### CODE REVIEW

File: packages/garden/src/lib/types/fieldSettings.ts [hunk 1]
```diff
@@ -28,7 +26,7 @@ export type FieldSetting<
  * @template ExtendedFields (optional) string literal that defines fields that does not exist on the base model.
  */
 export type FieldSettings<
-	ItemType,
-	ExtendedFields extends string = never,
-	TCustomGroupByKeys extends BaseRecordObject<TCustomGroupByKeys> = never
+	ItemType extends Record<PropertyKey, unknown>,
+	ExtendedFields extends string = string,
```
Comment (jaysencpp) [inline packages/garden/src/lib/types/fieldSettings.ts:30]: Litt vanskelig å se hva som skjer når man bytter fra default verdi "never" til "string", men tror tanken her var at dersom man ikke tar i bruk "ExtendedFields" så skal metodene i `fieldsSettings`-objektet som f.eks `getKey` med typene
```ts
¨getKey: (item: TData, itemKey: keyof TData | TExtendedFIelds, customGroupByKeys?: TCustomGroupByKeys)
``` 
få type `never` for `TExtendedFields`, men nå får den type = string (?)

File: packages/garden/src/lib/types/gardenOptions.ts [hunk 2]
```diff
@@ -1,8 +1,14 @@
-import React, { MemoExoticComponent, MutableRefObject } from 'react';
+import { MemoExoticComponent, MutableRefObject } from 'react';
 import { GardenController } from '../classes';
 import { GardenGroup, GardenGroups } from '.';
 
-export interface CustomItemView<TData, TExtendedFields extends string, TCustomGroupByKeys, TCustomState, TContext> {
+export interface CustomItemView<
+	TData extends Record<PropertyKey, unknown>,
+	TExtendedFields extends string = string,
+	TCustomGroupByKeys extends Record<PropertyKey, unknown> = Record<PropertyKey, unknown>,
```
Comment (jaysencpp) [inline packages/garden/src/lib/types/gardenOptions.ts:8]: Blir det rett typing når man bruker `Record<PropertyKey, unknown` her, men i GardenController blir det brukt
`TCustomGroupByKeys extends BaseRecordObject<TCustomGroupByKeys> = BaseRecordObject<unknown>,`

**Review by jaysencpp — CHANGES_REQUESTED** [review]

File: packages/garden/src/lib/types/gardenOptions.ts (hunk 2 above)
Comment (Gustav-Eikaas) [reply packages/garden/src/lib/types/gardenOptions.ts:8]: Det er samme type

Comment (jaysencpp) [reply packages/garden/src/lib/types/gardenOptions.ts:8]: Hvis de er identiske burde man velge én av dem, så er man konsekvent og blir litt lettere for andre å lese

Comment (jaysencpp) [reply packages/garden/src/lib/types/gardenOptions.ts:8]: Kan ta det i en annen PR

**Review by jaysencpp — APPROVED** [review]

**Pull request merged**